### PR TITLE
Add suffix on the exact_matches join

### PR DESCRIPTION
**Pull request by stuckoverflo**
To reproduce: Run a dataset that has a field whose name is in any of the following:

```
['loc_tuple', 'code', 'matched_barangay', 'matched_city_municipality',
       'matched_province', 'reg']
```

### CODE REVIEW

**Comment by marksteve** [comment]
@florobarotjr `row.get` will return `None` when there's a clash

**Comment by marksteve** [comment]
@florobarotjr PING

**Comment by florobarotjr** [comment]
@marksteve, made some changes to fix this inside the matcher. However, we'll still face the same issue on the final join (at the `check` part). If the client dataset has any of `matched_*`, `code`, and `total_score` in their dataset fields, they will be renamed to `*_linksight` since we have that part.

Hmm, still thinking of what the best approach here to deal with column names :/

**Comment by florobarotjr** [comment]
Rebased and set to point to `master`. Will continue working on this

File: linksight/api/matchers/ngrams_matcher.py [hunk 1]
```diff
@@ -175,10 +180,32 @@ def search_reference(self, search_tuple, ngram_table, nresults):
...
+            'dataset_{}'.format(old_columns.get('municity')): new_columns.get('municity'),
+            'dataset_{}'.format(old_columns.get('bgy')): new_columns.get('bgy')
+        }, inplace=True)
+        return dataset_df
+
     def get_matches(self):
         dataset_df = pd.read_csv(self.dataset_file)
 
-        columns = self.columns
+        dataset_df = self.prepend_dataset_columns(dataset_df)
```
Comment (marksteve) [inline linksight/api/matchers/ngrams_matcher.py]: No need to prefix since we're renaming?

File: linksight/api/matchers/ngrams_matcher.py [hunk 2]
```diff
@@ -175,10 +180,32 @@ def search_reference(self, search_tuple, ngram_table, nresults):
...
+        columns = self.dataset_columns
+        dataset_df = self.rename_interlevels(dataset_df, columns)
```
Comment (marksteve) [inline linksight/api/matchers/ngrams_matcher.py]: Remove other columns: `dataset_df[list(dataset_columns.values())]`

File: linksight/api/matchers/ngrams_matcher.py [hunk 3]
```diff
@@ -18,6 +18,11 @@
 
 class NgramsMatcher(BaseMatcher):
     reference = REFERENCE_FILE
+    dataset_columns = OrderedDict([
```
Comment (marksteve) [inline linksight/api/matchers/ngrams_matcher.py]: Better to define this only inside `get_matches`

**Pull request merged**